partiql-lang/src/main/kotlin/org/partiql/lang/planner/transforms/plan/PlanTyper.kt

```diff
@@ -334,12 +334,32 @@ internal object PlanTyper : PlanRewriter<PlanTyper.Context>() {
 
     override fun visitRexQueryScalarSubquery(node: Rex.Query.Scalar.Subquery, ctx: Context): PlanNode {
         val query = visitRex(node.query, ctx) as Rex.Query.Collection
-        when (val queryType = query.grabType() ?: handleMissingType(ctx)) {
+        // If it is SELECT VALUE, do not coerce.
+        if (query.constructor != null) {
+            val type = query.type as? CollectionType
+            return node.copy(query = query, type = type?.elementType?.flatten())
+        }
+        val type = when (val queryType = query.grabType() ?: handleMissingType(ctx)) {
             is CollectionType -> queryType.elementType
             else -> error("Query collection subqueries should always return a CollectionType.")
         }
+        val resultType = when (type) {
+            is StructType -> {
+                if (type.contentClosed && type.fields.size == 1) {
+                    type.fields[0].value
+                } else {
+                    handleCoercionError(ctx, type)
+                    StaticType.ANY
+                }
+            }
+            else -> {
```

Comment:
would we also need to consider a type w/ a union of structs for the coercion?

Reply:
The reason it isn't necessary to handle unions in this scenario is because we are specifically looking at the output of a scalar subquery -- which is SFW, which should always have a return type of `CollectionType(StructType(...))`.

Reply:
I see. Perhaps adding a comment that the return type will always be a collection type of structs could be helpful